### PR TITLE
 Add TUNING.md

### LINKED ISSUE #1133: Add tuning guide / documentation
Let's have a tuning.md

### PR DESCRIPTION
**Pull request by interwq**
View in https://github.com/interwq/jemalloc/blob/tuning/TUNING.md
This resolves #1133

### CODE REVIEW

**Comment by interwq** [comment]
@jasone : we can definitely use your input on the doc here (Please feel free to take over and make a new diff if you'd like).

**Comment by djwatson** [comment]
Are there some example numbers for muzzy_decay_ms/dirty ? It seems to beat around the bush a bit.

File: TUNING.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    time](http://jemalloc.net/jemalloc.3.html#arena.i.dirty_decay_ms) if
+    frequent reuse is expected.
+
+* [Extent hooks](http://jemalloc.net/jemalloc.3.html#arena.i.extent_hooks)
+
+    Extent hooks allow customization for managing underlying memory.  One use
+    case for performance purpose is to utilize huge pages -- for example,
+    [HHVM](https://github.com/facebook/hhvm/blob/master/hphp/util/alloc.cpp)
+    uses explicit arenas with customized extent hooks to manage 1GB huge pages
+    for frequently accessed data, which improves TLB misses significantly.
```
Comment (davidtgoldblatt) [inline TUNING.md]: I might say "which reduces TLB misses significantly.".

File: TUNING.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    Suggested: try `percpu_arena:percpu` or `percpu_arena:phycpu` if
+    thread migration between processors is expected to be infrequent.
+
+Examples:
+
+* High resource consumption application, prioritizing CPU utilization:
+
+    `background_thread:true,metadata_thp:auto` combined with relaxed decay time
+    (increased `dirty_decay_ms` and / or `muzzy_decay_ms`,
+    e.g. `dirty_decay_ms:30000,muzzy_decay_ms:30000`).
```
Comment (davidtgoldblatt) [inline TUNING.md]: I might even suggest turning off decay entirely if there's lots of spare memory.

**Comment by davidtgoldblatt** [comment]
A couple small suggestions (feel free to ignore if you disagree)

File: TUNING.md (hunk 1 above)
Comment (interwq) [reply TUNING.md]: Yep fixed.

File: TUNING.md (hunk 2 above)
Comment (interwq) [reply TUNING.md]: Feels a bit risky, so I'm going to leave this out for now. Maybe we can have a "extremely aggressive" case later if asked.

**Pull request merged**